pandas/core/arrays/base.py

```diff
@@ -682,6 +696,33 @@ def searchsorted(self, value, side="left", sorter=None):
         arr = self.astype(object)
         return arr.searchsorted(value, side=side, sorter=sorter)
 
+    def equals(self, other: "ExtensionArray") -> bool:
+        """
+        Return if another array is equivalent to this array.
+
+        Parameters
+        ----------
+        other: ExtensionArray
+            Array to compare to this Array.
+
+        Returns
+        -------
+        boolean
+            Whether the arrays are equivalent.
+
+        """
+        if not type(self) == type(other):
+            return False
```

Comment:
We should verify that this is the behavior we want. Namely

1. other array-likes are not equivalent, even if they are all equal.
2. *subclasses* are not equivalent, even if they are all equal.

The first seems fine. Not sure about the second.

Reply:
I was planning to open an issue about this after this PR (and keep it strict here), because this is right now a bit inconsistent within pandas, and might require a more general discussion / clean-up (eg Series.equals is more strict (requires same dtype) than Index.equals ...) 
But we can certainly also have the discussion here.

Reply:
Happy to be strict for now.

Reply:
I made it even more strict (same dtype, not just same class), and added a test for that. 
Will open an issue for the general discussion.

Reply:
I opened an issue for this at https://github.com/pandas-dev/pandas/issues/33940

In the end, it seems mainly to come to whether the dtype should exactly be equal or not. 

Since for EAs, the dtype is right now tied to the array class, using equal dtype for now also implies the same class (no additional check to allow sublcasses).